src/renderer/components/ReactFlowBox.tsx

```diff
@@ -135,19 +156,119 @@ const updateZIndexes = (
         }
     }
 
+    const nodesToAdjust = new Map<string, number>();
     // set the zIndex of all edges
     for (const e of edges) {
+        edgesById.set(e.id, e);
         let zIndex = Math.max(
-            byId.get(e.source)?.zIndex ?? STARTING_Z_INDEX,
-            byId.get(e.target)?.zIndex ?? STARTING_Z_INDEX
+            nodesById.get(e.source)?.zIndex ?? STARTING_Z_INDEX,
+            nodesById.get(e.target)?.zIndex ?? STARTING_Z_INDEX
         );
 
-        if (e.selected && zIndex < MIN_SELECTED_INDEX) {
-            zIndex += SELECTED_ADD;
+        const sourceNode = nodesById.get(e.source);
+        const targetNode = nodesById.get(e.target);
+
+        if (sourceNode && targetNode) {
+            if (e.selected && zIndex < MIN_SELECTED_INDEX) {
+                zIndex += SELECTED_ADD;
+                // If an edge is selected, make the connected nodes the same zIndex so that the handle stays above
+                if (sourceNode.type !== 'iterator') {
+                    // We need to wait until after the loop to adjust the zIndex
+                    nodesToAdjust.set(sourceNode.id, zIndex);
+                }
+                if (targetNode.type !== 'iterator') {
+                    // We need to wait until after the loop to adjust the zIndex
+                    nodesToAdjust.set(targetNode.id, zIndex);
+                }
+            } else if (sourceNode.selected || targetNode.selected) {
+                // If an edge is connected to a selected node, we need to make the other node it is connected to the same zIndex so that both handles stay above
+                if (targetNode.selected && targetNode.type !== 'iterator') {
+                    // We need to wait until after the loop to adjust the zIndex
+                    nodesToAdjust.set(sourceNode.id, zIndex);
+                }
+                if (sourceNode.selected && sourceNode.type !== 'iterator') {
+                    // We need to wait until after the loop to adjust the zIndex
+                    nodesToAdjust.set(targetNode.id, zIndex);
+                }
+            }
+            // Save this for later
+            const edgeParent = sourceNode.parentNode ?? targetNode.parentNode;
+            if (edgeParent) {
+                iteratorChildEdgesMap.set(edgeParent, [
+                    ...(iteratorChildEdgesMap.get(edgeParent) ?? []),
+                    e.id,
+                ]);
+            }
         }
 
         e.zIndex = zIndex - 1;
     }
+
+    // Adjust the zIndex of the nodes that were connected to selected edges
+    // The reason we don't do this in the loop above is because we don't want to modify as we detect)
+    for (const [nodeId, zIndex] of nodesToAdjust.entries()) {
+        const node = nodesById.get(nodeId);
+        if (node) {
+            node.zIndex = zIndex;
```

Comment:
I still don't see this being correct. What if one node is included with multiple different zIndexes in `nodesToAdjust`?